hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxySupport.java

```diff
@@ -280,25 +281,26 @@ private VersionedValue lockAndGet(Data key, long timeout) {
     }
 
     protected Set queryInternal(final Predicate predicate, final IterationType iterationType, final boolean dataResult) {
-        final NodeEngine nodeEngine = getNodeEngine();
+        NodeEngine nodeEngine = getNodeEngine();
         OperationService operationService = nodeEngine.getOperationService();
         Collection<MemberImpl> members = nodeEngine.getClusterService().getMemberList();
         int partitionCount = nodeEngine.getPartitionService().getPartitionCount();
         Set<Integer> partitions = new HashSet<Integer>(partitionCount);
         QueryResultSet result = new QueryResultSet(nodeEngine.getSerializationService(), iterationType, dataResult);
-        List<Integer> missingList = new ArrayList<Integer>();
+
         try {
             List<Future> futures = new ArrayList<Future>();
-            invokeQueryOperaration(predicate, operationService, members, futures);
+            invokeQueryOperation(predicate, operationService, members, futures);
             collectResults(partitions, result, futures);
             if (partitions.size() == partitionCount) {
                 return result;
             }
-            findMissingPartitions(partitionCount, partitions, missingList);
         } catch (Throwable t) {
-            missingList.clear();
-            findMissingPartitions(partitionCount, partitions, missingList);
+            EmptyStatement.ignore(t);
```

Comment:
Please have a look at this! I saw no sense in calling `findMissingPartitions()` as last statement in the `try` block and calling it again in the `catch` clause (after clearing the list). I think this can be safely moved out of that block.